### PR TITLE
`r\digital_twins_instance`: support `identity`

### LINKED ISSUE #15970: Support for identity parameter in azurerm_digital_twins_instance
### Description
There is no option to specify _identity_ for the **azurerm_digital_twins_instance**

### New or Affected Resource(s)
* azurerm_digital_twins_instance

### Potential Terraform Configuration

```
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example_resources"
  location = "West Europe"
}

resource "azurerm_digital_twins_instance" "example" {
  name                = "example-DT"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  tags = {
    foo = "bar"
  }

 identity {
    type = "SystemAssigned"
  }
}
````

### References
* https://docs.microsoft.com/en-us/azure/digital-twins/concepts-security
* https://docs.microsoft.com/en-us/azure/digital-twins/how-to-route-with-managed-identity?tabs=portal#add-a-system-managed-identity-during-instance-creation
* https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/digital_twins_instance

### PR DESCRIPTION
**Pull request by myc2h6o**
Fix #15970

### CODE REVIEW

File: internal/services/digitaltwins/digital_twins_instance_resource_test.go [hunk 1]
```diff
@@ -181,3 +215,30 @@ resource "azurerm_digital_twins_instance" "test" {
...
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  identity {
+    type = "SystemAssigned"
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r DigitalTwinsInstanceResource) identityNone(data acceptance.TestData) string {
```
Comment (tombuildsstuff) [inline internal/services/digitaltwins/digital_twins_instance_resource_test.go:234]: we can use `basic` for this purpose?

**Review by tombuildsstuff — APPROVED** [review]
One comment since the test case can be reused, but this otherwise LGTM 👍

File: internal/services/digitaltwins/digital_twins_instance_resource_test.go (hunk 1 above)
Comment (myc2h6o) [reply internal/services/digitaltwins/digital_twins_instance_resource_test.go]: @tombuildsstuff yes, I've updated the test code to reuse the basic config

**Comment by myc2h6o** [comment]
Updated test result
image

**Pull request merged**